### PR TITLE
Server refactoring

### PR DESCRIPTION
**Pull request by hpelin**
Server code has been heavily refactored. The code is now modular, hence the new files were created:
- models.py - here, the model classes are defined. Currently, the CellposeModel is implemented. Pending - segment anything model.
- segmentationclasses.py - here, the types of segmentation are defined. The idea is that we do not use different branches for different projects, but rather define the respective classes here.
- serviceclasses.py - bentoml service class defined. If in the future we decide to use a different one, its class definition would go here
- fsimagestorage.py - same logic as in the client - everything related to scikit image and image processing, loading and saving is there
- bentomlrunners.py - custom runners needed for bentoml service
- config.cfg - JSON type file with all the configuration - including the training and evaluation arguments. If we wish to change anything, we do it here and the rest is taken care of automatically. E.g. if you wish to run Mito project segmentation  you just change the "segmentation" type in the config file. Or if you wish to increase the number of epochs you do that in the config as well.

Modifications:
- settings file is no longer there, replaced by config.cfg
- Readme file of the full directory is changed with the description of the config file.
- training is enabled - reads from the path and the trained model is saved in the path provided in config file.

The code runs and masks are created. However, the more detailed testing is needed, especially when parsing the arguments into parent classes (e.g. original cellposemodel)

### CODE REVIEW

File: src/server/dcp_server/bentomlrunners.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        :param img: image to evaluate on
+        :type img: np.ndarray
+        :param z_axis: z dimension (optional, default is None)
+        :type z_axis: int
+        :return: mask of the image, list of 2D arrays, or single 3D array (if do_3D=True) labelled image.
+        :rtype: np.ndarray
+        """              
+
+        mask, _, _ = self.model.eval(img=img, z_axis=z_axis)
```
Comment (christinab12) [inline src/server/dcp_server/bentomlrunners.py:37]: maybe this return is too cellpose specific?

File: src/server/dcp_server/models.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+from cellpose import models, utils
+from segment_anything import SamPredictor, sam_model_registry
```
[the remark this answers is not in the document]
Comment (christinab12) [reply src/server/dcp_server/models.py]: commented for now

File: src/server/dcp_server/bentomlrunners.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        return mask
+
+    @bentoml.Runnable.method(batchable=False)
```
[the remark this answers is not in the document]
Comment (hpelin) [reply src/server/dcp_server/bentomlrunners.py:41]: Added as an issue
